### DISCUSSION TITLE
API for custom functions for guests ride selection.

### DISCUSSION THREAD

**Discussion by Nrgte** [OpenRCT2/OpenRCT2#21958] [Feature requests]
It would be nice if plugins could use an API to overwrite how guests determine which ride they want to go on or which item they want to purchase. This would allow modders to introduce new guest stats such as how stingy they are or how weather tolerant they are or how long of a queue they tolerate.

**Comment by autosysops** [comment]
With these settings in the guest interface you can already manipulate where peeps want to go

```
        /**
         * The guest's minimum preferred intensity between 0 and 15.
         */
        minIntensity: number;

        /**
         * The guest's maximum preferred intensity between 0 and 15.
         */
        maxIntensity: number;

        /**
         * The guest's tolerance to nauseating rides between 0 and 3.
         */
        nauseaTolerance: number;
```

And there are also values for thirst and hunger etc. So I kind of feel this is already available.
